modules/twinkleprod.js

```diff
@@ -91,11 +119,7 @@ Twinkle.prod.callback = function twinkleprodCallback() {
 	var evt = document.createEvent( "Event" );
 	evt.initEvent( 'change', true, true );
 	result.prodtype[0].dispatchEvent( evt );
```

Comment:
This is throwing an error outside mainspace (aka files and books).  I think part of the issue is that you're trying to make it smarter about the notification (lines 93-97) but haven't refactored `Twinkle.prod.callback.prodtypechanged` (lines 125+).

Reply:
This was supposed to execute only on articles. Fixed now by wrapping it in a if (namespace==='article'). 
Didn't notice this earlier because the PROD nom still goes on smoothly in any namespace. There are no visible problems, barring the error shown in the console.

Reply:
I meant it more to try to understand the reason for duplicating the code in `Twinkle.prod.callback.prodtypechanged`.  AFAICT, the "Notify page creator if possible" stuff you added for files and books just repeats what's in that function.  I think it's nice to ditch the fake event if possible (maybe even the hiding of the fieldset?!) but am I missing a reason to do it this way as opposed to just falling back to the `prod` case?

Reply:
There can be two options:
1. Generate the prod-type fieldset on all pages, and then hide it on non-articles. This avoid duplication of the code that generates the notify checkbox and reason textarea.
2. Generate the fieldset only on articles. This mean no duplication of code. But on non-articles, this scheme is more efficient as the fieldset isn't generated in the first place.
 
It originally was (1) but I changed it to (2) in the 2nd commit. On further thought, the minor efficiency gain, that too only on non-articles, is probably not worth the code duplication. So I have reverted it back to (1), as you also seem to be suggesting. Is this fine?

Ditch the fake event? This seems pretty standard across TW modules. I don't see why or how it can be ditched.

Reply:
Awesome, thanks.  I'll get around to a final test tomorrow — library internet miserable today — but that looks good to me.

>Ditching
I had thought that was your intention.